parlai/core/torch_classifier_agent.py

```diff
@@ -213,21 +212,33 @@ def _get_labels(self, batch):
             labels_tensor = labels_tensor.cuda()
         return labels_tensor
 
-    def _update_confusion_matrix(self, batch, predictions):
+    def _update_confusion_matrix(self, predictions, batch):
         """
         Update the confusion matrix given the batch and predictions.
 
-        :param batch:
-            a Batch object (defined in torch_agent.py)
         :param predictions:
             (list of string of length batchsize) label predicted by the
             classifier
+        :param batch:
+            a Batch object (defined in torch_agent.py)
         """
-        for i, pred in enumerate(predictions):
-            label = batch.labels[i]
-            self.metrics['confusion_matrix'][(label, pred)] += 1
+        for class_name in self.class_list:
+            if not self.opt['get_all_metrics']:
+                class_name = self.ref_class
+            prec_str = 'class_{}_prec'.format(class_name)
+            recall_str = 'class_{}_recall'.format(class_name)
+            f1_str = 'class_{}_f1'.format(class_name)
+            weighted_f1_str = 'class_{}_weighted_f1'.format(class_name)
+            precision, recall, f1 = ClassificationMetric.compute_metrics(
+                predictions, batch.labels, class_name
+            )
+            self.record_local_metric(prec_str, precision)
+            self.record_local_metric(recall_str, recall)
+            self.record_local_metric(f1_str, f1)
+            if not self.opt['get_all_metrics']:
```

Comment:
HERE. HERE's what's missing.

We should be creating a WeightedF1Metric HERE and doing record_local_metrics!

Reply:
No, this won't work. bc you should pass the already aggregated ```f1```  metric? Here is btz size f1 right?